lib/railway_ipc/message_consumption.ex

```diff
@@ -0,0 +1,159 @@
+defmodule RailwayIpc.MessageConsumption do
```

Comment:
This module handles consumption flow for both event and command message consumption. Is this too confusing? Is there a better way to split it up? @StevenNunez